target/consensource-api
# feat: add search param

Signed-off-by: Patrick-Erichsen <Patrick.Erichsen@target.com>

## Proposed change/fix

Adds full-text search capabilities on the `search` param on `/factories` that will match on the name of the factory, and the standard name of certifications that the factory holds.

Also fixes a bug where the `total` field on the response was incorrect due to not being filtered on Factory orgs.

## Types of changes

What types of changes does this pull request introduce to ConsenSource? _Put an `x` in the boxes that apply_


## How to run/test

#### In the `consensource-compose` repo

- `./docker-helper.sh -b api`

#### In the `api` repo

Manual Testing
- `./docker-helper -r api`
- Create a factory through the UI (or CLI)
- Create a cert following [these instructions](https://github.com/target/consensource-ui/pull/56)
- Verify each response returns the factory:
  - `curl localhost:3000/search?search=<factory_name>`
  - `curl localhost:3000/search?search=<cert_std_name>`
  - `curl localhost:3000/search?search=<factory_address_field>`

Integration Tests
- Tear down compose network for the app
- `cd test/`
- `docker-compose -f docker-compose.dev.yaml up`
- `docker exec -it test-rest-api /bin/bash`
- `cargo test route_handlers::factories::tests -- --nocapture --test-threads=1`
- Verify that all 6 tests pass
